streams/src/main/java/org/apache/kafka/streams/processor/internals/ProcessorTopology.java

```diff
@@ -197,4 +197,14 @@ public String toString(final String indent) {
         return sb.toString();
     }
 
+    // for testing only
+    public Set<String> processorConnectedStateStores(final String processorName) {
+        for (ProcessorNode<?, ?> node : processorNodes) {
+            if (node.name().equals(processorName)) {
+                return node.stateStores;
```

Comment:
although only public for testing maybe return `Collections.unmodifiableSet(new HashSet<>(node.stateStores));` just as a precaution?